x-pack/plugins/ml/public/application/components/anomalies_table/links_menu.tsx

```diff
@@ -57,33 +58,54 @@ interface LinksMenuProps {
 
 export const LinksMenuUI = (props: LinksMenuProps) => {
   const [openInDiscoverUrl, setOpenInDiscoverUrl] = useState<string | undefined>();
+  const [discoverUrlError, setDiscoverUrlError] = useState<string | undefined>();
 
   const isCategorizationAnomalyRecord = isCategorizationAnomaly(props.anomaly);
 
   const closePopover = props.onItemClick;
 
   const kibana = useMlKibana();
+  const {
+    services: { share, application },
+  } = kibana;
 
   useEffect(() => {
     let unmounted = false;
 
     const generateDiscoverUrl = async () => {
-      const {
-        services: { share },
-      } = kibana;
       const discoverLocator = share.url.locators.get('DISCOVER_APP_LOCATOR');
 
       if (!discoverLocator) {
-        // eslint-disable-next-line no-console
-        console.error('No locator for Discover detected');
+        const discoverLocatorMissing = i18n.translate(
+          'xpack.ml.anomaliesTable.linksMenu.discoverLocatorMissingErrorMessage',
+          {
+            defaultMessage: 'No locator for Discover detected',
+          }
+        );
+
+        setDiscoverUrlError(discoverLocatorMissing);
+
         return;
       }
 
       const job = mlJobService.getJob(props.anomaly.jobId);
 
       const index = job.datafeed_config.indices[0];
       const interval = props.interval;
-      const dataViewId = (await getDataViewIdFromName(index)) || index;
+      const dataViewId = await getDataViewIdFromName(index);
+
+      // If data view doesn't exist for some reasons
+      if (!dataViewId) {
+        const autoGeneratedDiscoverLinkError = i18n.translate(
+          'xpack.ml.anomaliesTable.linksMenu.autoGeneratedDiscoverLinkErrorMessage',
+          {
+            defaultMessage: `Unable to generate link to Discover because no data view exists for index '{index}'`,
```

Comment:
Not mandatory, but IMO you can drop a few words here:
```suggestion
            defaultMessage: `Unable to link to Discover; no data view exists for index '{index}'`,
```

Reply:
Updated here https://github.com/elastic/kibana/pull/124408/commits/c69569426a81963480a6978c335d6e0bccd4194c